src/main/java/net/dv8tion/jda/api/entities/Entitlement.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.entities;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+
+/**
+ * Represents a Discord Entitlement for premium App subscriptions.
+ * <br>This should contain all information provided from Discord about an {@link Entitlement Entitlement}.
+ */
+public interface Entitlement extends ISnowflake
+{
+    /**
+     * The id of the entitlement
+     *
+     * @return The id of the {@link Entitlement Entitlement}
+     */
+    @Override
+    long getIdLong();
+
+    /**
+     * The id of the SKU related to this {@link Entitlement Entitlement}
+     *
+     * @return The id of the SKU related to this {@link Entitlement Entitlement}
+     */
+    long getSkuIdLong();
+
+    /**
+     * The id of the SKU related to this {@link Entitlement Entitlement}
+     *
+     * @return The id of the SKU related to this {@link Entitlement Entitlement}
+     */
+    @Nonnull
+    String getSkuId();
+
+    /**
+     * The id of the parent application of this {@link Entitlement Entitlement}
+     *
+     * @return The id of the parent application of this {@link Entitlement Entitlement}
+     */
+    long getApplicationIdLong();
+
+    /**
+     * The id of the parent application of this {@link Entitlement Entitlement}
+     *
+     * @return The id of the parent application of this {@link Entitlement Entitlement}
+     */
+    @Nonnull
+    String getApplicationId();
+
+    /**
+     * The id of the user that purchased the {@link Entitlement Entitlement}
+     *
+     * @return The id of the user that purchased the {@link Entitlement Entitlement}
+     */
+    @Nullable
+    Long getUserIdLong();
+
+    /**
+     * The id of the user that purchased the {@link Entitlement Entitlement}
+     *
+     * @return The id of the user that purchased the {@link Entitlement Entitlement}
+     */
+    @Nullable
+    String getUserId();
```

Comment:
This should be default-implemented like in https://github.com/discord-jda/JDA/blob/7bdeea8612dd3cc6537d9ef211de4f3c790bf872/src/main/java/net/dv8tion/jda/api/entities/ISnowflake.java#L36-L40